### DISCUSSION TITLE
Integrating with Suspense

### DISCUSSION THREAD

**Discussion by elge70** [next-safe-action/next-safe-action#74] [General]
Currently   const { execute, result, status } = useAction(server_action); 
is working great. Is it possible to use integrate it with Suspense ?

If we wrap the component inside suspense, it loads the loading.tsx but, immediately it vanishes. The status gives the status accurately.
if the sever action gives an error, status and result shows the error but not moved to Errorboundary.

**Comment by TheEdoRan** [comment]
> if the sever action gives an error, status and result shows the error but not moved to Errorboundary.

This should be solved by using `handleReturnedServerError` to rethrow server errors. Check out [this comment](https://github.com/TheEdoRan/next-safe-action/issues/79#issuecomment-1987336593) I left on a related issue.
